### PR TITLE
ci: add container runtime gate

### LINKED ISSUE #30: Prepare CI gates for safe Renovate automerge
## Problem

The repository has Renovate enabled, but its default-branch ruleset does not require any CI status checks. The current build workflow builds the image, but it does not expose a stable aggregate gate and still contains Dependabot-specific behavior even though dependency automation is moving to Renovate. A successful image build alone also does not verify that the packaged plugins can be copied by the entrypoint as intended.

## Proposed outcome

- Add an always-present, stable functional CI gate for pull requests.
- Keep a real container build for Dockerfile, plugin-list, downloader, and entrypoint changes, without publishing from pull requests.
- Add a lightweight runtime smoke test that starts the built image with a writable plugin volume and verifies that the expected non-empty plugin artifacts are copied to the Graylog plugin directory.
- Use explicit change detection so unrelated documentation or workflow-only changes do not run expensive multi-platform builds while still receiving a successful aggregate gate.
- Remove Dependabot-specific conditions and validate Renovate-only behavior.
- Keep lint and link checks deterministic through the shared reliability workstreams.
- Configure strict required status checks separately from the human-review rule, with no Renovate bypass on CI.
- Enable repository auto-merge and Renovate automerge for the agreed non-major update types only after positive, red-check, stale-base, and fork pull-request canaries pass.

## Acceptance criteria

- Every pull request receives stable required gate names.
- Build, download, or entrypoint smoke failures and cancellations block the functional gate.
- Irrelevant changes skip expensive work but still produce successful aggregate gates.
- Pull requests never publish images and fork pull requests cannot access write tokens or repository secrets through validation jobs.
- Human-authored pull requests still require approval.
- Eligible Renovate pull requests merge without human approval only when the branch is current and all required checks are green.
- A deliberately failing required check and a stale branch both prevent automerge.

## Related work

- Parent rollout: Netcracker/qubership-monitoring-operator#499
- Super-Linter reliability: Netcracker/qubership-monitoring-operator#496
- Link validation reliability: Netcracker/qubership-monitoring-operator#497
- Relevant execution of expensive jobs: Netcracker/qubership-monitoring-operator#498

This issue is repository-specific. Reusable improvements discovered here should be proposed to the appropriate shared repository without making this rollout depend on an unapproved shared implementation.

### PR DESCRIPTION
**Pull request by IldarMinaev**
## Why

The repository builds and publishes a plugin init image, but its pull-request workflow does not expose a stable functional gate. The existing build also does not verify that the entrypoint copies non-empty JAR files into the mounted Graylog plugin directory, and it disables TLS certificate verification while downloading plugins.

## What

- Add fail-closed path detection and an always-present `Graylog Plugins Init CI Gate` for pull requests.
- Build a nonpublishing `linux/amd64` test image and verify that every configured plugin is copied as a non-empty top-level JAR file.
- Keep registry login, package write access, and multi-platform publication limited to release events and relevant pushes to `main`.
- Download plugins only from HTTPS GitHub release URLs with TLS certificate verification enabled.
- Add downloader regression coverage and update Super-Linter to the v8.7.0 compatibility baseline.

## How to verify

- Run `tests/test_download_plugins.sh`.
- Run `shellcheck download_plugins.sh entrypoint.sh tests/test_download_plugins.sh`.
- Run `actionlint .github/workflows/build.yaml .github/workflows/super-linter.yaml`.
- Run `docker build --check .` and `docker build --no-cache -t qubership-graylog-plugins-init:ci .`.
- Start the image with a writable volume mounted at `/usr/share/graylog/plugin` and verify that both configured JAR files are copied directly into the volume.

Closes #30

### CODE REVIEW

File: .github/workflows/super-linter.yaml [hunk 1]
```diff
@@ -89,10 +89,9 @@ jobs:
         fi
 
     - name: Lint Code Base
-      uses: super-linter/super-linter/slim@ffde3b2b33b745cb612d787f669ef9442b1339a6 # v8.1.0
+      uses: super-linter/super-linter/slim@4ce20838b8ab83717e78138c5b3a1407148e0918 # v8.7.0
```
Comment (asatt) [inline .github/workflows/super-linter.yaml]: ### [blocking] Super-Linter v8.7.0 fails BIOME_FORMAT on `.qubership/grand-report.json`

Confidence: high

- Problem: The bump to v8.7.0 makes `BIOME_FORMAT` report `Found 2 errors.` for `.qubership/grand-report.json`, so `run-lint` is red on this head (run 32478830373). The same file passes under v8.1.0 on `main` and on #32, so the bump is the trigger, and the PR cannot merge behind a red lint check.
- Proposed solution: Reformat `.qubership/grand-report.json` to Biome's output, or exclude the generated report from `BIOME_FORMAT` in `.github/super-linter.env`, so `run-lint` passes on this revision.

File: entrypoint.sh [hunk 2]
```diff
@@ -9,9 +9,9 @@ GRAYLOG_PLUGINS_DIR=/usr/share/graylog/plugin
 echo "Start graylog-plugins-init container..."
 
 echo "Available plugins:"
-ls -lah ${LOCAL_PLUGINS_DIR}
+ls -lah "${LOCAL_PLUGINS_DIR}"
 
 echo "Copy plugins from ${LOCAL_PLUGINS_DIR} to ${GRAYLOG_PLUGINS_DIR} (which should be mounted as a volume)..."
-cp -r "${LOCAL_PLUGINS_DIR}" "${GRAYLOG_PLUGINS_DIR}"
+cp -r "${LOCAL_PLUGINS_DIR}/." "${GRAYLOG_PLUGINS_DIR}/"
```
Comment (asatt) [inline entrypoint.sh]: ### [blocking] The entrypoint and downloader fix duplicates open PR #32

Confidence: high

- Problem: This PR rewrites `cp -r "${LOCAL_PLUGINS_DIR}" ...` to `cp -r "${LOCAL_PLUGINS_DIR}/." ...` and replaces the `wget -P` destination in `download_plugins.sh`; #32 changes the same lines of the same two files for the same two bugs, with different wording. Whichever merges first leaves the other with a conflict, and the runtime fix is separable from the CI gate this PR is about.
- Proposed solution: Decide which PR owns the runtime fix, drop those hunks from the other, and rebase it.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -1,76 +1,201 @@
...
+      - name: Test plugin downloader
+        run: tests/test_download_plugins.sh
+      - name: Build test image
+        uses: docker/build-push-action@53b7df96c91f9c12dcc8a07bcb9ccacbed38856a # v7.3.0
+        with:
+          context: .
           file: Dockerfile
+          load: true
+          no-cache: true
+          platforms: linux/amd64
```
Comment (asatt) [inline .github/workflows/build.yaml]: ### [non-blocking] Pull requests no longer build linux/arm64

Confidence: high

- Problem: The previous workflow built `linux/amd64,linux/arm64` on every pull request with `push: false`; the smoke test builds `linux/amd64` only, while `publish_image` still publishes both platforms. An arm64-only build failure now surfaces at publish time on `main` or on a release, not in review.
- Proposed solution: Either add a non-loading `linux/arm64` build to `runtime_smoke`, or state in the PR description that arm64 build coverage moves to the publish job.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -1,76 +1,201 @@
...
+        uses: actions/checkout@3d3c42e5aac5ba805825da76410c181273ba90b1 # v7.0.1
+        with:
+          fetch-depth: 100
+          persist-credentials: false
+      - name: Detect changed paths
+        id: filter
+        if: github.event_name != 'release'
+        uses: dorny/paths-filter@ceb8a2b8f2d89434be7ff52d3de7ec3738c5cc9d # v4.0.3
+        with:
+          base: ${{ github.ref }}
```
Comment (asatt) [inline .github/workflows/build.yaml]: ### [non-blocking] The `base` input to paths-filter has no effect

Confidence: high

- Problem: `dorny/paths-filter` ignores `base` on `pull_request` when a token is present — and `token` defaults to `${{ github.token }}` — logging a warning on every PR run. On `push` to `main`, `github.ref` shortens to `main`, which equals the head ref and yields the default comparison against `payload.before`.
- Proposed solution: Remove the `base` input.

File: tests/test_download_plugins.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+https://github.com/example/plugin/releases/download/1.0.0/plugin-1.0.0.jar
+EOF
+run_downloader "$valid_list" "$TEST_ROOT/valid-downloads"
+test -s "$TEST_ROOT/valid-downloads/plugin-1.0.0.jar"
+
+invalid_source_list="$TEST_ROOT/invalid-source.list"
+cat >"$invalid_source_list" <<'EOF'
+https://example.com/plugin-1.0.0.jar
+EOF
+if run_downloader "$invalid_source_list" "$TEST_ROOT/invalid-source-downloads"; then
```
Comment (asatt) [inline tests/test_download_plugins.sh]: ### [non-blocking] The negative downloader test does not assert the failure reason

Confidence: high

- Problem: The test only requires `download_plugins.sh` to exit non-zero for a non-GitHub URL. A regression that aborts the script earlier — an unbound variable under `set -u`, a broken `PLUGINS_FILE` default — keeps the test green while the host allowlist is no longer enforced.
- Proposed solution: Capture stderr and assert that it contains `Unsupported plugin URL:`, and add a case covering the empty-download guard.

**Review by asatt — CHANGES_REQUESTED** [review]
Result: REQUEST_CHANGES (3 blocking, 3 non-blocking)
Revision: e6c735e52c449e0fa3b1a34d42445b324f111a2a...cb41f70353c17544ff4a936b0a276afe92246508

## Findings

### [blocking] Editing README.md exposes a NATURAL_LANGUAGE failure on line 18

Confidence: high

`README.md:18` (reported here because the line is outside the diff and cannot take an inline comment)

- Problem: Touching `README.md` adds it to Super-Linter's changed-file set, and `textlint` then fails with `18:14  error  Incorrect term: “docker”, use “Docker” instead  terminology` in run 32478830373. The line is pre-existing and untouched by this PR, but it now makes `run-lint` red.
- Proposed solution: Change `docker image` to `Docker image` on line 18 so the linted file is clean.

Model: Opus 5

File: .github/workflows/super-linter.yaml (hunk 1 above)
Comment (denifilatoff) [reply .github/workflows/super-linter.yaml]: Fixed: `.qubership/grand-report.json` is now Biome's `{}`. `renovate.json` uses tabs so `BIOME_FORMAT` stays green after the v8.7.0 bump.

File: entrypoint.sh (hunk 2 above)
Comment (denifilatoff) [reply entrypoint.sh]: #32 merged on 2026-09-04 (`cp -r "${LOCAL_PLUGINS_DIR}/."` plus `mkdir -p`). This branch is rebased onto that. Remaining downloader changes are the TLS-verified wget, empty-file guard, and tests — they were not in #32 (main still uses `--no-check-certificate`).

File: .github/workflows/build.yaml (hunk 3 above)
Comment (denifilatoff) [reply .github/workflows/build.yaml]: Fixed: `runtime_smoke` still loads `linux/amd64` for the copy-contract test and now runs a non-loading `linux/arm64` build so an arm64-only Dockerfile break fails in review, not only on publish.

File: .github/workflows/build.yaml (hunk 4 above)
Comment (denifilatoff) [reply .github/workflows/build.yaml]: Fixed: removed the `base` input so `paths-filter` uses its default comparison and stops warning on pull requests.

File: tests/test_download_plugins.sh (hunk 5 above)
Comment (denifilatoff) [reply tests/test_download_plugins.sh]: Fixed: the negative case now requires stderr to contain `Unsupported plugin URL:`, and a second case covers the empty-download guard (`Downloaded plugin is empty:`).

**Comment by denifilatoff** [comment]
Review note on README terminology: `README.md:18` now says `Docker image`, which was the blocking NATURAL_LANGUAGE finding on the previous head.

**Review by asatt — APPROVED** [review]
